src/sql/azdata.proposed.d.ts

```diff
@@ -462,6 +477,16 @@ declare module 'azdata' {
 			 */
 			None = 'None'
 		}
+
+		/**
+		 * Opens the change password dialog in connection management service.
+		 */
+		export function openChangePasswordDialog(initialConnectionProfile: IConnectionProfile): void;
+
+		/**
+		 * Gets the connection profile and params from the last error (required to avoid circular dependencies)
+		 */
+		export function getConnectionProfileFromError(): Thenable<ConnectionProfile>;
```

Comment:
The profile seems like something that should be passed along with the handleError call. (or probably better, the connection ID).

There's no guarantee that by the time that method is called that this is going to actually be able to return the correct error (what happens when there are multiple connections happening at once? That's a race condition waiting to happen)

Reply:
The problem is that searching the profile using the connectionID brings up nothing when attempting to connect using the connection ID provided in the Connection Status Manager (I checked it when it was running and it is empty).

Additionally, attempting to pass in the connection profile directly resulted in a circular dependency happening: 

<img width="482" alt="Circular Error additional parameters" src="https://user-images.githubusercontent.com/18018452/214164235-69201d77-5495-4461-93e3-db836c57099e.png">

Reply:
1. The profile will exist *somewhere* - if anything you clearly have the profile when starting the connection so when an error occurs you should be able to just pass it directly from there

2. That looks like you're trying to pass a full object (with injected services) over JSON-RPC, which isn't allowed.

Instead you need to ensure that anything passed over that layer are converted into JSON-RPC-safe versions of the object before transport, and then rehydrated back into full objects on the other side if needed.

Reply:
I suggest that I add a mapping system instead to handle this